src/MPQEditor/MPQEditor.ts

```diff
@@ -267,6 +269,91 @@ export class MPQEditorProvider implements vscode.CustomTextEditorProvider {
     //TODO process messages
   }
 
+  /**
+   * @brief Get path of the parent .circle file
+   */
+  private getModelFilePath(document: vscode.TextDocument): string {
+    const dirPath = path.parse(document.uri.path).dir;
+    let fileName =
+      this._mpqDataMap[document.uri.toString()].getSection("model_path");
+    return path.join(dirPath, fileName);
+  }
+
+  /**
+   * @brief Get all model nodes
+   */
+  private handleRequestModelNodes(
+    document: vscode.TextDocument,
+    webview: vscode.Webview
+  ): void {
+    const K_DATA: string = "data";
+    const K_EXIT: string = "exit";
+    const K_ERROR: string = "error";
+    let modelFilePath = this.getModelFilePath(document);
+
+    // TODO integrate with Toolchain
+    const tool = "/usr/share/one/bin/circle-operator";
+    if (!fs.existsSync(tool)) {
+      // check whether it is installed
+      Balloon.info(
+        "To add more layers for editing please install ONE-toolchain"
```

Comment:
@jyoungyun , is `ONE-toolchain` name appropriate? or if not, can you please suggest?

Reply:
> @stamalakhov How about using just `Toolchain`? ONE-vscode supports two types of toolchains, ONE and TRIV. And each toolchain contains `one-compiler` so user can use `onecc` command in any toolchain.

Ok. I see. I'll change it to `Toolchain`. Thank you.

Reply:
@seanshpark @jyoungyun 
done.